ansibullbot/triagers/plugins/docs_info.py

```diff
@@ -0,0 +1,294 @@
+                                content[:ds_span.start()].splitlines()
+                            )
+                            func_map.ds_line_end = (
+                                func_map.ds_line_start + len(func_doc.splitlines())
+                        )
+                    class_map.funcs.append(func_map)
+
+            mod_map.classes.append(class_map)
+
+    return mod_map
+
+def _is_diff_docs_only(file_content, diff):
+    """ Check a python file's changes to see if they're only docstring
+        changes.
+    """
+
+    diff = _get_diff_info(diff)
+    source = None
+
+    if file_content is not None:
+        source = _get_ast_info(file_content)
+
+    if source is None:
+        return False
+
+    for line in diff:
+        # Check if change applies to module DOCUMENTATION (if exists)
+        if source.doc_string:
+            if source.ds_line_start <= line["lineno"] <= source.ds_line_end:
+                continue
+
+        # Check if change applies to module EXAMPLES (if exists)
+        if source.example_string:
+            if source.ex_line_start <= line["lineno"] <= source.ex_line_end:
+                continue
+
+        # Find appropriate class if it exists
+        source_class = source.find_class(line["lineno"])
+        if source_class is not None:
+            if source_class.doc_string:
+                if source_class.ds_line_start <= line["lineno"] <= source_class.ds_line_end:
+                    continue
+
+            source_func = source_class.find_function(line["lineno"])
+            if source_func is not None and source_func.doc_string:
+                if source_func.ds_line_start <= line["lineno"] <= source_func.ds_line_end:
+                    continue
+
+        # If we made it this far, this change is outside docs/examples
+        return False
+
+    return True
+
+def _is_docs_only(changed_file):
+    """ Check if the changes made to ``changed_file`` affect only documentation. """
+
+    if isinstance(changed_file, dict):
+        changed_file = CommitFile(changed_file)
+
+    if _is_docs_path(changed_file.filename):
+        return True
+
+    # Additions or deletions of complete files outside of
+    # docsite/examples will likely be more than documentation
+    # changes
+    if changed_file.status != "modified":
+        return False
+
+    # Non-Python files that are outside of the docsite
+    # folder will not be documentation related
+    if not changed_file.filename.endswith(".py"):
+        return False
+
+    # If a python file, check if the changes are only to
+    # docstrings
+    if _is_diff_docs_only(changed_file.file_content, changed_file.patch):
+        return True
+
+    return False
+
+def get_docs_facts(iw):
+    """ Cycle through the files and gather facts about documentation changes. """
+    dfacts = {
+        "is_docs_only": False
+    }
+
+    if not iw.is_pullrequest():
+        return dfacts
+
+    docs_only = False
+
+    pr_files = iw.pullrequest.get_files()
```

Comment:
Please us `iw.files` at it uses cache.

Reply:
`iw.files`, as it is currently written, only returns a list of the filenames. It does not return the full list of `self.pr_files`, with each item containing the attributes we subsequently use.

My previous changes in https://github.com/ansible/ansibullbot/pull/1505/commits/19e77980360c9bc982c6a70c605434267d36730a utilized the cached approach, while leaving `iw.files` as-is.

Reply:
Ah, didn't realize that. Then using `iw.load_update_fetch_files()` is what we want here.

And sorry for changing those methods while you were working on this PR.

Reply:
> And sorry for changing those methods while you were working on this PR.

No need to be sorry. The cleanup was a good thing, and `get_files` was broken and unused.

> Then using `iw.load_update_fetch_files()` is what we want here.

I'll work towards using `load_update_fetch_files` directly. The mocking in `IssueMock` is doable, but could be confusing for future uses as it would be completely different in functionality.

Do we maybe want to re-introduce `get_files`? Or, as my brain over-complicates things, make `self.pr_files` a property:

```py
class DefaultWrapper:
    def __init__(self, ...):
        #self.pr_files = None -->
        self._pr_files = None

    @property
    def pr_files(self):
        if self._pr_files is None:
            self._pr_files = self.load_update_fetch('files')
        return self._pr_files

    @property
    def files(self):
        if self.is_issue():
            return None
        return [x.filename for x in self.pr_files]
```

Reply:
I am fine with the above addition.